src/dswx_sar/mosaic_rtc_burst.py

```diff
@@ -10,26 +10,156 @@
 from dswx_sar import dswx_sar_util
 
-logger = logging.getLogger('dswx_S1')
+logger = logging.getLogger('dswx_s1')
 
-def check_reprojection(geogrid_mosaic,
-                       rtc_image: str,
-                       nlooks_image: str = None) -> bool:
+
+def majority_element(num_list):
+
+    """
+    Determine the majority element in a list using the
+    Boyer-Moore Voting Algorithm.
+
+    Parameters:
+    -----------
+    num_list : List[int]
+        A list of integers where the majority element needs to be determined.
+
+    Returns:
+    -----------
+    int:
+        The majority element in the list. If no majority exists,
+        it may return any element from the list.
+    """
+    idx, ctr = 0, 1
+
+    for i in range(1, len(num_list)):
+        if num_list[idx] == num_list[i]:
+            ctr += 1
+        else:
+            ctr -= 1
+            if ctr == 0:
+                idx = i
+                ctr = 1
+
+    return num_list[idx]
+
+
+def read_metadata_epsg(h5_meta_path):
+    '''
+    Extract metadata regarding coordinate spacing and projection from an HDF5 file.
+
+    parameters:
+    -----------
+    h5_meta_path: str
+        The path to the HDF5 file containing the metadata.
+
+    returns:
+    -------
+    meta_dict : dict
+        A dictionary containing:
+        - 'xspacing': Spacing along the x-coordinate.
+        - 'yspacing': Spacing along the y-coordinate.
+        - 'epsg': The EPSG code for the data projection.
+
+    '''
+    freqA_path = '/data/'
+    with h5py.File(h5_meta_path, 'r') as src_h5:
+        xres = np.array(src_h5[f'{freqA_path}/xCoordinateSpacing'])
+        yres = np.array(src_h5[f'{freqA_path}/yCoordinateSpacing'])
+        epsg = np.array(src_h5[f'{freqA_path}/projection'])
+
+    meta_dict = {}
+    meta_dict['xspacing'] = xres
+    meta_dict['yspacing'] = yres
+    meta_dict['epsg'] = epsg
+    return meta_dict
+
+def save_h5_metadata_to_tif(h5_meta_path,
+                            data_path,
+                            output_tif_path,
+                            epsg_output):
+    '''
+    extract data from RTC metadata and store it as geotiff
+    paremeters:
+    -----------
+    h5_meta_path: str
+        the path to the rtc metadata
+    data_path: str
+        the hdf5 path to the data to be extracted
+    output_tif_path: str
+        output tif path
+    '''
+    freqA_path = '/data/'
+
+    with h5py.File(h5_meta_path, 'r') as src_h5:
+        data = np.array(src_h5[data_path])
+        xcoord = np.array(src_h5[f'{freqA_path}/xCoordinates'])
+        ycoord = np.array(src_h5[f'{freqA_path}/yCoordinates'])
+        xres = np.array(src_h5[f'{freqA_path}/xCoordinateSpacing'])
+        yres = np.array(src_h5[f'{freqA_path}/yCoordinateSpacing'])
+        epsg = np.array(src_h5[f'{freqA_path}/projection'])
+
+    dtype = data.dtype
+
+    geotransform = [xcoord[0], float(xres), 0,
+                    ycoord[0], 0, float(yres)]
```

Comment:
```suggestion
    geotransform = [xcoord[0] - xres / 2, float(xres), 0,
                    ycoord[0] - yres / 2, 0, float(yres)]
```

There will be a half pixel offset between the first values of [`xcoord`, `ycoord`] and the upper left corner in the RTC products geotransformation (in geotiff). I would suggest to either apply the half pixel offset (like suggestion above), or load those geotransformation information from RTC layer in geotiff.